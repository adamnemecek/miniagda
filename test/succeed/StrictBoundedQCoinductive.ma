-- 2010-11-26

data Bool : Set
{ true : Bool
; false : Bool
}

let C : Size -> Set
      = \ i -> [j : Size] -> |j| < |i| -> Bool

cofun foo : [i : Size] -> C i
{ foo ($i) j = true
}

{- does not type check
cofun loop : [i : Size] -> C i
{ loop ($i) j = loop i j
}
-}
