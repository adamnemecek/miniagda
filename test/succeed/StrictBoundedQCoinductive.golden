MiniAgda by Andreas Abel and Karl Mehltretter
--- opening "StrictBoundedQCoinductive.ma" ---
--- scope checking ---
--- type checking ---
type  Bool : Set
term  Bool.true : < Bool.true : Bool >
term  Bool.false : < Bool.false : Bool >
type  C : Size -> Set
type  C = \ i -> .[j : Size] -> |j| < |i| -> Bool
term  foo : .[i : Size] -> C i
{ foo $[i < #] [j] = Bool.true
}
--- evaluating ---
--- closing "StrictBoundedQCoinductive.ma" ---
